api/v1alpha1/dspipeline_types.go

```diff
@@ -342,6 +342,10 @@ type GRPC struct {
 	Image     string                `json:"image,omitempty"`
 	// +kubebuilder:validation:Optional
 	Port string `json:"port"`
+	// +kubebuilder:validation:Optional
+	CertificateContents string `json:"certificateContents,omitempty"`
+	// +kubebuilder:validation:Optional
+	PrivateKeyContents string `json:"privateKeyContents,omitempty"`
```

Comment:
I don't love that the user needs to paste these in here. For apiserver, we are able to just use the paths to the OpenShift-provded certs like so:
https://github.com/opendatahub-io/data-science-pipelines-operator/blob/e101b0fc722048a6bcbdf4f2a7efa69281e71b73/config/internal/apiserver/default/deployment.yaml.tmpl#L180-L191

could we not do something similar for here?

Reply:
This is automatically set by the operator and is not for users to use (technically they can). I had to add these fields to be able to set the contents in the template.
Regarding setting the contents, I also don't like it, but the MLMD config requires that.

Reply:
re: having to set the contents -- sure, no way around that.

> I had to add these fields to be able to set the contents in the template.

Ok, I understand the reason. I really don't like adding fields to the API for this though. Hmm. How sure are we that there's no other way to feed things to the manifestival templating? That doesn't sound right to me.

Reply:
yeah I think you could just add two strings roughly here
https://github.com/opendatahub-io/data-science-pipelines-operator/blob/20dd747d4342088a42dfce3cc98f907573ea3f82/controllers/dspipeline_params.go#L70

and then populate them

and then use those to feed the templating engine

and avoid changing the DSPA API

Reply:
Done.

Reply:
looks perfect!